src/shader_recompiler/ir/passes/resource_tracking_pass.cpp

```diff
@@ -656,6 +656,32 @@ void PatchImageInstruction(IR::Block& block, IR::Inst& inst, Info& info, Descrip
     }
     ASSERT(image.GetType() != AmdGpu::ImageType::Invalid);
     const bool is_storage = IsImageStorageInstruction(inst);
+
+    // Patch image instruction if image is fmask.
+    if (image.IsFmask()) {
+        // Ignore fmask writes. TODO: handle dimension queries.
+        if (is_storage && inst.GetOpcode() == IR::Opcode::ImageQueryDimensions) {
```

Comment:
In case of dimensions query, better to put assert too or at least emit some error log message

Reply:
I will implement that before merge